src/index.js

```diff
@@ -1,32 +1,41 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const API = 'https://rickandmortyapi.com/api/character/';
 
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+function fetchData(url_api) {
+  return new Promise((resolve, reject) => {
+    let XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest;
+    let xhttp = new XMLHttpRequest();
+    xhttp.open('GET', url_api, true);
+    xhttp.send();
+    xhttp.onreadystatechange = () => {
+      if (xhttp.readyState === 4) {
+        if (xhttp.status == 200) {
+          resolve(JSON.parse(xhttp.responseText));
+        } else {
+          reject(url_api);
```

Comment:
Te recomiendo ser mas claros en el reject, esto te permite identificar el error mas fácil. 

```suggestion
          reject(new Error("Algo salió mal."));
```